cmake: Update pmdk version to 1.7

CongMinYin opened the pull request:
<!--
Thank you for opening a pull request!  Here are some tips on creating
a well formatted contribution.

Please give your pull request a title like "[component]: [short description]"

This is the format for commit messages:

"""
[component]: [short description]

[A longer multiline description]

Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
Signed-off-by: [Your Name] <[your email]>
"""

The Signed-off-by line is important, and it is your certification that
your contributions satisfy the Developers Certificate or Origin.  For
more detail, see SubmittingPatches.rst.

The component is the short name of a major daemon or subsystem,
something like "mon", "osd", "mds", "rbd, "rgw", etc. For ceph-mgr modules,
give the component as "mgr/<module name>" rather than a path into pybind.

For more examples, simply use "git log" and look at some historical commits.

This was just a quick overview.  More information for contributors is available here:
https://raw.githubusercontent.com/ceph/ceph/master/SubmittingPatches.rst

-->
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

CongMinYin commented:
@tchaikov Currently I just used PMDK repository in the code. Should I continue to use ceph/nvml repository?  In fact the nvml is renamed to pmdk. Should I create a PMDK reposity in ceph?

tchaikov commented on cmake/modules/Buildpmem.cmake:
```diff
@@ -12,8 +12,8 @@ function(build_pmem)
   ExternalProject_Add(nvml_ext
-      GIT_REPOSITORY "https://github.com/ceph/nvml.git"
-      GIT_TAG "dd622819dd4ee97d3920f913c70be"
+      GIT_REPOSITORY "https://github.com/pmem/pmdk.git"
```
s/pmem/ceph/ see https://github.com/ceph/pmdk

tchaikov replied:
and please note the rationale in the commit message, for instance

1. nvml is renamed to pmdk, and
2. we need to bump to the latest stable release: 1.7 ..

lixiaoy1 replied:
> s/pmem/ceph/ see https://github.com/ceph/pmdk

@tchaikov Hi Kefu, is this what you want to update PMDK repository in Ceph? https://github.com/ceph/pmdk/pull/2

tchaikov replied:
as discussed offline. i just sync'ed ceph/pmdk. so we just need to point `GIT_REPOSITORY` to it and update `GIT_TAG` to `1.8`.

dillaman replied:
+1 to use named tag `stable-1.8` if possible instead of commit hash

tchaikov replied:
@CongMinYin could you take this opportunity to s/nvml_ext/pmdk_ext/ as well?

CongMinYin replied:
> @CongMinYin could you take this opportunity to s/nvml_ext/pmdk_ext/ as well?

@tchaikov Yes, I am testing it now.

dillaman commented on cmake/modules/Buildpmem.cmake:
```diff
@@ -12,8 +12,8 @@ function(build_pmem)
-      GIT_TAG "dd622819dd4ee97d3920f913c70be"
+      GIT_REPOSITORY "https://github.com/pmem/pmdk.git"
+      GIT_TAG "bc5e309485df61c452d08367e4b13ba9dfed5071"
       SOURCE_DIR ${CMAKE_BINARY_DIR}/src/nvml
```
Should we be updating all the `nvml` directory references to `pmdk` as well?

CongMinYin commented:
@tchaikov 
① nvml is renamed pmdk，changed the nvml repository to pmdk repository ,
② update pmdk GIT_TAG to 1.7
③ renamed the directory "nvml" to "pmdk"

tchaikov commented:
@CongMinYin i mean, the external project is now named "nvml_ext", could you rename it to "pmem_ext" in this change?

CongMinYin commented:
> @CongMinYin i mean, the external project is now named "nvml_ext", could you rename it to "pmem_ext" in this change?
@tchaikov  Yes, I mean I renamed it to "pmdk_ext" and do test now. Should I renamed it to "pmem_ext"?

tchaikov commented:
> > @CongMinYin i mean, the external project is now named "nvml_ext", could you rename it to "pmem_ext" in this change?
> 
> @tchaikov Yes, I mean I renamed it to "pmdk_ext" and do test now. Should I renamed it to "pmem_ext"?

oh, i see. and yeah, i agree with you. `pmdk_ext` is better.

CongMinYin commented:
> > > @CongMinYin i mean, the external project is now named "nvml_ext", could you rename it to "pmem_ext" in this change?
> > 
> > 
> > @tchaikov Yes, I mean I renamed it to "pmdk_ext" and do test now. Should I renamed it to "pmem_ext"?
> 
> oh, i see. and yeah, i agree with you. `pmdk_ext` is better.

@tchaikov I have renamed "nvml_ext " to "pmdk_ext" and tested.
